src/Storefront/Page/Product/Review/ProductReviewsLoadedEvent.php

```diff
@@ -13,120 +13,68 @@
-         */
-        protected EntitySearchResult $searchResult;
-
-        /**
-         * @var SalesChannelContext
-         */
-        protected $salesChannelContext;
-
-        /**
-         * @var Request
-         */
-        protected $request;
-
-        /**
-         * @param EntitySearchResult<ProductReviewCollection> $searchResult
-         */
-        public function __construct(
-            EntitySearchResult $searchResult,
-            SalesChannelContext $salesChannelContext,
-            Request $request
-        ) {
-            $this->searchResult = $searchResult;
-            $this->salesChannelContext = $salesChannelContext;
-            $this->request = $request;
-        }
-
-        /**
-         * @return EntitySearchResult<ProductReviewCollection>
-         */
-        public function getSearchResult(): EntitySearchResult
-        {
-            return $this->searchResult;
-        }
-
-        public function getSalesChannelContext(): SalesChannelContext
-        {
-            return $this->salesChannelContext;
-        }
-
-        public function getContext(): Context
-        {
-            return $this->salesChannelContext->getContext();
-        }
-
-        public function getRequest(): Request
-        {
-            return $this->request;
-        }
+/**
+ * @deprecated tag:v6.7.0 - Use \Shopware\Core\Content\Product\SalesChannel\Review\Event\ProductReviewsLoadedEvent instead
+ */
+#[Package('storefront')]
+class ProductReviewsLoadedEvent extends NestedEvent implements ShopwareSalesChannelEvent
+{
+    /**
+     * @var StorefrontSearchResult<ProductReviewCollection>
+     */
+    protected $searchResult;
+
+    /**
+     * @var SalesChannelContext
+     */
+    protected $salesChannelContext;
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @param StorefrontSearchResult<ProductReviewCollection> $searchResult
+     */
+    public function __construct(
+        StorefrontSearchResult $searchResult,
+        SalesChannelContext $salesChannelContext,
+        Request $request
+    ) {
+        $this->searchResult = $searchResult;
+        $this->salesChannelContext = $salesChannelContext;
+        $this->request = $request;
     }
-} else {
-    #[Package('storefront')]
-    class ProductReviewsLoadedEvent extends NestedEvent implements ShopwareSalesChannelEvent
-    {
-        /**
-         * @deprecated tag:v6.7.0 - Type will change to EntitySearchResult<ProductReviewCollection>
-         *
-         * @var StorefrontSearchResult<ProductReviewCollection>
-         */
-        protected $searchResult;
 
-        /**
-         * @var SalesChannelContext
-         */
-        protected $salesChannelContext;
+    /**
+     * @return StorefrontSearchResult<ProductReviewCollection>
+     */
+    public function getSearchResult(): StorefrontSearchResult
+    {
+        Feature::triggerDeprecationOrThrow('v6.7.0.0', CoreProductReviewsLoadedEvent::class);
```

Comment:
if you only call this, the second parameter needs a message, which you can generate with `deprecatedMethodMessage` or `deprecatedClassMessage` but with out it, now the message would only be the class name :wink:

Reply:
Ah yes sorry I missed that. Now changed.